File: innn/intersect-android/app/src/main/java/com/tmmt/innersect/mvp/model/Commodity.java
package com.tmmt.innersect.mvp.model;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by flame on 2017/9/14.
 */

public class Commodity implements Parcelable{

    public String skuName;
    public long productId;
    //SkuInfo skuInfo;
    public int quantity;
    public String skuSize;
    public float salePrice;
    public String skuColor;
    public String skuThumbnail;
    public String salesStatus;

    public  int colorValId;

    public long id;

    public int ascount;//

    public int aas; //0 no
    public String asdesc;
    public int isas; //0不可以

    public String getName() {
        if (skuName == null) {
            return "no name";
        }
        String name = skuName.replaceAll("\\\\n", " ");
        return name;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(skuName);
        dest.writeLong(productId);
        dest.writeString(skuColor);
        dest.writeString(skuSize);
        dest.writeString(skuThumbnail);
        dest.writeInt(ascount);
        dest.writeLong(id);
    }

    public Commodity(){}

    public Commodity(Parcel in){
        skuName=in.readString();
        productId=in.readLong();
        skuColor=in.readString();
        skuSize=in.readString();
        skuThumbnail=in.readString();
        ascount=in.readInt();
        id=in.readLong();
    }


    public static final Parcelable.Creator<Commodity> CREATOR = new Parcelable.Creator<Commodity>()
    {
        public Commodity createFromParcel(Parcel in)
        {
            return new Commodity(in);
        }
        public Commodity[] newArray(int size)
        {
            return new Commodity[size];
        }
    };

    public boolean isPreSale() {
        return salesStatus.equalsIgnoreCase("presell");
    }

//    public String getSkuDate() {
//        if (skuInfo != null && skuInfo.sku_date != null) {
//            return skuInfo.sku_date.replace(",", "");
//        }
//        return " ";
//    }
//    static class SkuInfo{
//        String sku_date;
//    }
}

File: app/src/main/java/com/stesso/android/model/Commodity.kt
//package com.stesso.android.model
//
//import android.os.Parcel
//import android.os.Parcelable
//
///**
// * Created by flame on 2017/9/14.
// */
//
//class Commodity : Parcelable {
//
//    var skuName: String? = null
//    var productId: Long = 0
//
//    var quantity: Int = 0
//    var skuSize: String? = null
//    var salePrice: Float = 0.toFloat()
//    var skuColor: String? = null
//    var skuThumbnail: String? = null
//    var salesStatus: String? = null
//
//    var colorValId: Int = 0
//
//    var id: Long = 0
//
//    var ascount: Int = 0//
//
//
//    override fun writeToParcel(dest: Parcel, flags: Int) {
//        dest.writeString(skuName)
//        dest.writeLong(productId)
//        dest.writeString(skuColor)
//        dest.writeString(skuSize)
//        dest.writeString(skuThumbnail)
//        dest.writeInt(ascount)
//        dest.writeLong(id)
//    }
//
//    constructor() {}
//
//    constructor(`in`: Parcel) {
//        skuName = `in`.readString()
//        productId = `in`.readLong()
//        skuColor = `in`.readString()
//        skuSize = `in`.readString()
//        skuThumbnail = `in`.readString()
//        ascount = `in`.readInt()
//        id = `in`.readLong()
//    }
//
//    companion object {
//
//
//        val CREATOR: Parcelable.Creator<Commodity> = object : Parcelable.Creator<Commodity> {
//            override fun createFromParcel(`in`: Parcel): Commodity {
//                return Commodity(`in`)
//            }
//
//            override fun newArray(size: Int): Array<Commodity> {
//                return arrayOfNulls(size)
//            }
//        }
//    }
//
//}
